File: android/src/main/java/com/videotrim/interfaces/VideoTrimListener.java
package com.videotrim.interfaces;

import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.bridge.WritableMap;
import com.videotrim.enums.ErrorCode;

import java.util.Map;

public interface VideoTrimListener {
  void onLoad(int duration);
  void onTrimmingProgress(int percentage);
  void onFinishTrim(String url, long startMs, long endMs, int videoDuration);
  void onCancelTrim();
  void onError(String errorMessage, ErrorCode errorCode);
  void onCancel();
  void onSave();
  void onLog(WritableMap log);
  void onStatistics(WritableMap statistics);
}

File: android/src/main/java/com/videotrim/interfaces/VideoTrimListener.kt
package com.videotrim.interfaces

import com.facebook.react.bridge.WritableMap
import com.videotrim.enums.ErrorCode

interface VideoTrimListener {
  fun onLoad(duration: Int)
  fun onTrimmingProgress(percentage: Int)
  fun onFinishTrim(url: String, startMs: Long, endMs: Long, videoDuration: Int)
  fun onCancelTrim()
  fun onError(errorMessage: String?, errorCode: ErrorCode)
  fun onCancel()
  fun onSave()
  fun onLog(log: WritableMap)
  fun onStatistics(statistics: WritableMap)
}
